### PR TITLE
Fx some issues with a11y

### PR DESCRIPTION
**Pull request by edlerd**
## Done

- fixed a11y issue with labels for selectable table
- fixed heading hierarchy for empty state
- fixed column heading for snapshots table

Fixes [list issues/bugs if needed]

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @lorumic or @edlerd for access.
    - With a local copy of this branch, run as described in the [Readme](https://github.com/canonical/lxd-ui#setting-up-for-development).
2. Perform the following QA steps:
    - test the changed pages are still good.

### CODE REVIEW

File: src/components/SelectableMainTable.tsx [hunk 1]
```diff
@@ -59,12 +59,11 @@ const SelectableMainTable: FC<Props> = ({
       content: (
         <>
           <CheckboxInput
-            label=""
+            label={<div className="u-off-screen">Select all</div>}
             labelClassName="multiselect-checkbox"
             checked={isAllSelected}
             indeterminate={isSomeSelected && !isAllSelected}
             onChange={isSomeSelected ? selectNone : selectPage}
-            aria-label="Select all"
```
Comment (lorumic) [inline src/components/SelectableMainTable.tsx:67]: Why was this removed here but not below? (line 115)

File: src/components/SelectableMainTable.tsx [hunk 2]
```diff
@@ -108,12 +107,13 @@ const SelectableMainTable: FC<Props> = ({
       {
         content: (
           <CheckboxInput
-            label=""
+            label={
+              <div className="u-off-screen">`Select ${row.name ?? ""}`</div>
```
[the remark this answers is not in the document]
Comment (edlerd) [reply src/components/SelectableMainTable.tsx]: Yes! Actually row.name ?? "row", I fixed it just now.

**Review by lorumic — APPROVED** [review]
